### PR TITLE
[collection] Add `TimingWheel.HasKey()` method and test

### LINKED ISSUE #2918: [collection] Add `TimingWheel.HasKey()` method
`TimingWheel{}` uses a unique key to identify a task.

However, it doesn't have a `HasKey()` method to check whether a key is existed.

```go
func (tw *TimingWheel) SetTimer(key, value any, delay time.Duration) error
```

## Solution

Create a `HasKey()` method.

## Alternative

The user could track the existence of keys by maintaining an external copy of `timers *SafeMap`.

But naturally it's not natural, wasting memory, and has concurrency issue.

### PR DESCRIPTION
**Pull request by ofey404**
Closes #2918 . See the issue for more description.

`TimingWheel.HasKey()` is a natural requirement, to check whether a task does exist or not.

### CODE REVIEW

File: core/collection/timingwheel.go [hunk 1]
```diff
@@ -162,6 +162,11 @@ func (tw *TimingWheel) Stop() {
 	close(tw.stopChannel)
 }
 
+func (tw *TimingWheel) HasKey(key any) bool {
```
Comment (kevwan) [inline core/collection/timingwheel.go:165]: Contains?

Comment (ofey404) [reply core/collection/timingwheel.go:165]: Well, `Contains(key) bool` is okay, but `HasKey(key) bool` would be more straightforward.

Especially when the argument isn't named as `key`, eg:

```go
var id string

// ...

tw.Contains(id)  // What contains ID? ID is contained as a task? key? value?

// vs

tw.HasKey(id)  // It's clear that timing wheel has ID as a key.
```

**Comment by ofey404** [comment]
> All operations on timers are within the run loop on a specific goroutine.
> 
> Even SafeMap is thread-safe, but it's for the golang issue. It might be removed later when the issue is fixed.

I noticed that all other `timers.Get()` sit in the `run()` method.

So, what change should we make to this PR? IMO since `SafeMap` is thread safe, the current implementation is acceptable.

**Comment by kevwan** [comment]
Yes, but safemap is for an go issue, which is mentioned in code comments. If the following issue got fixed, we'll use builtin map.

https://github.com/golang/go/issues/20135

**Comment by kevwan** [comment]
I think the following code meet your requirements?

```go
keys := make(map[string]struct{})
// add task keys

tw, err := collection.NewTimingWheel(time.Second, 600, func(key, value interface{}) {
	// synchronize
	delete(keys, key)
	job(&count)
})
```

**Comment by ofey404** [comment]
> I think the following code meet your requirements?
> 
> 
> 
> ```go
> 
> keys := make(map[string]struct{})
> 
> // add task keys
> 
> 
> 
> tw, err := collection.NewTimingWheel(time.Second, 600, func(key, value interface{}) {
> 
> 	// synchronize
> 
> 	delete(keys, key)
> 
> 	job(&count)
> 
> })
> 
> ```

I've already used this implementation in my scenario.

I got it: you don't recommend add this method, because it would access timers out of that specific goroutine.

After all I just wonder why? Performance issue?

**Comment by ofey404** [comment]
Oh, I misunderstand it, Golang map is not thread-safe.

[How safe are Golang maps for concurrent Read/Write operations? | StackOverflow](https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations)

Thank you for your guidance and I'll close this PR.

**Pull request closed without merging**